File: app/java/src/main/java/com/example/production/domain/model/cost/StandardCost.java
package com.example.production.domain.model.cost;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 標準原価マスタ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StandardCost {
    private Integer id;
    private String itemCode;
    private LocalDate effectiveStartDate;
    private LocalDate effectiveEndDate;
    private BigDecimal standardMaterialCost;
    private BigDecimal standardLaborCost;
    private BigDecimal standardExpense;
    private BigDecimal standardManufacturingCost;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Cost/StandardCost.cs
namespace ProductionManagement.Domain.Models.Cost;

/// <summary>
/// 標準原価マスタ
/// </summary>
public class StandardCost
{
    public long? Id { get; init; }

    public required string ItemCode { get; init; }

    public required DateOnly EffectiveStartDate { get; init; }

    public DateOnly? EffectiveEndDate { get; init; }

    public required decimal StandardMaterialCost { get; init; }

    public required decimal StandardLaborCost { get; init; }

    public required decimal StandardExpense { get; init; }

    public required decimal StandardManufacturingCost { get; init; }

    public DateTime CreatedAt { get; init; }

    public DateTime UpdatedAt { get; init; }
}
